app/screens/home/channel_list/servers/index.tsx

```diff
@@ -0,0 +1,162 @@
+        let unread = false;
+        let mentions = 0;
+        subscriptions.forEach((value) => {
+            unread = unread || value.unread;
+            mentions += value.mentions;
+        });
+        setTotal({mentions, unread});
+    };
+
+    const unreadsSubscription = (serverUrl: string, myChannels: MyChannelModel[]) => {
+        const unreads = subscriptions.get(serverUrl);
+        if (unreads) {
+            let mentions = 0;
+            let unread = false;
+            myChannels.forEach((myChannel) => {
+                mentions += myChannel.mentionsCount;
+                unread = unread || myChannel.isUnread;
+            });
+
+            unreads.mentions = mentions;
+            unreads.unread = unread;
+            subscriptions.set(serverUrl, unreads);
+            updateTotal();
+        }
+    };
+
+    const serversObserver = async (servers: ServersModel[]) => {
+        registeredServers.current = servers;
+        servers.forEach((server) => {
+            const {lastActiveAt, url} = server;
+            if (lastActiveAt && url !== currentServerUrl) {
+                const sdb = DatabaseManager.serverDatabases[url];
+                if (sdb?.database) {
+                    if (!subscriptions.has(url)) {
+                        const unreads: UnreadSubscription = {
+                            mentions: 0,
+                            unread: false,
+                        };
+                        subscriptions.set(url, unreads);
+                        unreads.subscription = sdb.database.
+                            get(MY_CHANNEL).
+                            query(Q.on(CHANNEL, Q.where('delete_at', Q.eq(0)))).
+                            observeWithColumns(['mentions_count', 'has_unreads']).
+                            subscribe(unreadsSubscription.bind(undefined, url));
+                    }
+                }
+
+                // subscribe and listen for unreads and mentions
+            } else if (subscriptions.has(url)) {
+                // logout from server, remove the subscription
+                subscriptions.delete(url);
+            }
+        });
+    };
+
+    const onPress = useCallback(() => {
+        if (registeredServers.current?.length) {
+            const renderContent = () => {
+                return (
+                    <ServerList servers={registeredServers.current!}/>
+                );
+            };
+
+            const snapPoints = ['50%', 10];
+            if (registeredServers.current.length > 3) {
+                snapPoints[0] = '90%';
+            }
+
+            const closeButtonId = 'close-your-servers';
+            bottomSheet({
+                closeButtonId,
+                renderContent,
+                snapPoints,
+                theme,
+                title: intl.formatMessage({id: 'servers.create_button', defaultMessage: 'Add a Server'}),
+            });
+        }
+    }, [isTablet, theme, registeredServers.current]);
```

Comment:
It seems that depending on `registeredServers.current` wont modify the behaviour in any way. `registeredServers` is a reference, so this function will always have the current value in current.